src/bin/cql-stress-cassandra-stress/settings/option/node.rs

```diff
@@ -0,0 +1,152 @@
+    param::{ParamsParser, SimpleParamHandle},
+    ParsePayload,
+};
+
+pub struct NodeOption {
+    pub nodes: Vec<String>,
+    pub whitelist: bool,
+    pub datacenter: Option<String>,
+}
+
+impl NodeOption {
+    pub const CLI_STRING: &str = "-node";
+
+    pub fn description() -> &'static str {
+        "Nodes to connect to"
+    }
+
+    pub fn parse(cl_args: &mut ParsePayload) -> Result<Self> {
+        let params = cl_args.remove(Self::CLI_STRING).unwrap_or(vec![]);
+        let (parser, handles) = prepare_parser();
+        parser.parse(params)?;
+        parse_with_handles(handles)
+    }
+
+    pub fn print_help() {
+        let (parser, _) = prepare_parser();
+        parser.print_help();
+    }
+
+    pub fn print_settings(&self) {
+        println!("Node:");
+        println!("  Nodes: {:?}", self.nodes);
+        println!("  Is White List: {}", self.whitelist);
+        println!("  Datacenter: {:?}", self.datacenter);
+    }
+}
+
+struct NodeParamHandles {
+    datacenter: SimpleParamHandle,
+    whitelist: SimpleParamHandle,
+    file: SimpleParamHandle,
+    nodes: SimpleParamHandle,
+}
+
+fn prepare_parser() -> (ParamsParser, NodeParamHandles) {
+    let mut parser = ParamsParser::new(NodeOption::CLI_STRING);
+
+    let datacenter = parser.simple_param(
+        "datacenter=",
+        r"^.*$",
+        None,
+        "Datacenter used for DCAwareRoundRobinLoadPolicy",
+        false,
+    );
+    let whitelist = parser.simple_param(
+        "whitelist",
+        r"^$",
+        None,
+        "Limit communications to the provided nodes",
+        false,
+    );
+    let file = parser.simple_param("file=", r"^.*$", None, "Node file (one per line)", false);
+    let nodes = parser.simple_param(
+        "",
+        r"^[^=,]+(,[^=,]+)*$",
+        Some("localhost"),
+        "comma delimited list of nodes",
+        false,
+    );
+
+    // $ ./cassandra-stress help -node
+    // Usage: -node [datacenter=?] [whitelist] [file=?] []
+    parser.group(vec![&datacenter, &whitelist, &file, &nodes]);
+
+    (
+        parser,
+        NodeParamHandles {
+            datacenter,
+            whitelist,
+            file,
+            nodes,
+        },
+    )
+}
+
+fn read_nodes_from_file(filename: String) -> Result<Vec<String>> {
+    let file = File::open(filename).context("Invalid nodes file")?;
+    let buf = io::BufReader::new(file);
+    Ok(buf
+        .lines()
+        .map(|l| l.unwrap())
+        .filter(|s| !s.is_empty())
+        .collect())
+}
+
+fn parse_with_handles(handles: NodeParamHandles) -> Result<NodeOption> {
+    let datacenter = handles.datacenter.get();
+    let whitelist = handles.whitelist.supplied_by_user();
+    let file = handles.file.get();
+    let nodes = handles
+        .nodes
+        .get()
+        .map(|list| list.split(',').map(|nd| nd.to_owned()).collect())
+        .unwrap();
+
+    let nodes = match file {
+        Some(f) => read_nodes_from_file(f)?,
+        None => nodes,
+    };
```

Comment:
> We demand that a user provide a `nodes` param (because of the `unwrap()` at line 112)

The user doesn't need to provide it. We `unwrap` because this parameter has the default value of `localhost`.